File: problemas/estruturas-e-bibliotecas/1069/1069_2.py
from collections import deque

N = int(input())

for _ in range(N):
    expressao = input().strip()

    resposta = 0
    pilha = deque()
    for letra in expressao:
        if(letra == '<'):
            pilha.append('<')
        elif(letra == '>' and len(pilha) > 0):
            resposta += 1
            pilha.pop()

    print(resposta)

File: problemas/estruturas-e-bibliotecas/1069/1069_2.cs
using System;
using System.Collections.Generic;

class URI {
    static void Main(string[] args) {
        int N = int.Parse(Console.ReadLine());
        
        for(int i = 0; i < N; ++i){
            int resposta = 0;
            Stack<int> pilha = new Stack<int>();
            string expressao = Console.ReadLine();

            for(int j = 0; j < expressao.Length; ++j){
                if(expressao[j] == '<'){
                    pilha.Push(expressao[j]);
                }else if(expressao[j] == '>' && pilha.Count > 0){
                    pilha.Pop();
                    resposta += 1;
                }
            }

            Console.WriteLine(resposta);
        }
    }
}